subprojects/maven/src/main/java/org/gradle/api/publish/maven/internal/publication/DefaultMavenPomDependencies.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.publish.maven.internal.publication;
+
+import org.gradle.api.publish.maven.internal.dependencies.MavenDependencyInternal;
+
+import java.util.Set;
+
+/**
+ * Default implementation of {@link MavenPomDependencies}.
+ */
+public class DefaultMavenPomDependencies implements MavenPomDependencies {
+
+    private final Set<MavenDependencyInternal> runtimeDependencies;
+    private final Set<MavenDependencyInternal> apiDependencies;
+    private final Set<MavenDependencyInternal> optionalDependencies;
+    private final Set<MavenDependencyInternal> runtimeDependencyConstraints;
+    private final Set<MavenDependencyInternal> apiDependencyConstraints;
+    private final Set<MavenDependencyInternal> importDependencyConstraints;
+
+    public DefaultMavenPomDependencies(
+        Set<MavenDependencyInternal> runtimeDependencies,
+        Set<MavenDependencyInternal> apiDependencies,
+        Set<MavenDependencyInternal> optionalDependencies,
```

Comment:
❗ I'd like to try to merge https://github.com/gradle/gradle/pull/23566 first, which will establish a second set of optional dependencies to do optional runtime vs. optional compile deps properly.  This will require some (hopefully easy) work here, splitting this collection to make the new tests in my PR continue to pass.

Reply:
We can merge that first. This PR is on hold for now 'cause of a bunch of problems with the signing infrastructure eagerly realizing publication artifacts.